File: accounting/accounting/report/balance_sheet/balance_sheet.py
# Copyright (c) 2013, ruthra and contributors
# License: MIT. See LICENSE

import frappe
from accounting.accounting.doctype.accounts.utils import calculate_balance_on_node, get_all_child_as_list

def execute(filters=None):
        columns = [
                {
                        'fieldname': 'account',
                        'fieldtype': 'Data',
                        'read_only': 1
                },
                {
                        'fieldname': 'balance',
                        'fieldtype': 'Currency',
                        'read_only': 1
                }]

        data = []
        accounts = frappe.db.get_list('Accounts',filters={'name':['in',['Assets','Liabilities']]}, fields=['name','type','is_group'])

        for acc in accounts:
                for node in get_all_child_as_list(acc):
                        data.append({'account': node['name'],'balance': calculate_balance_on_node(node)['balance']})

        return columns, data

File: accounting/accounting/report/balance_sheet/balance_sheet.js
// Copyright (c) 2016, ruthra and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Balance Sheet"] = {
    "filters": [

    ],
    "formatter": function(value, row, column, data, default_formatter){
	if(column.fieldtype == 'Currency' && !(value === undefined)){
	    console.log('value' in window);
	    value = new Intl.NumberFormat('en-IN', {style: 'currency', currency:'INR'}).format(value);
	    return "<div style='text-align: right'>"+ value + "</div>";
	}
	return default_formatter(value, row, column, data);
    }
};
